### PR TITLE
Update make docs python ci

### PR DESCRIPTION
**Pull request by marble**
There a three todos:
1. Check file domainindex.html - it has just bee copied and seems to work
2. CI.yml: Make CDN upload work
3. CI.yml: Make PyPi upload work.

Changes:
* b1b3367 Point out TODOs in workflow
* 97ec036 Tweak workflow
* e067a41 Use 'domainindex.html' of Sphinx basic theme
* dd2d094 Use requirements_dev.txt
* b9bf65f Make stat/ a Python module
* 2a9d070 Add documentation
* 2ab99f6 Update .gitignore and build/.gitkeep
* 3f999a7 Update Python setup and module
* 7ceea6e Add Makefile
* 5fc06de Switch to workflow CI.yml
* fb0d6db Use jobfile-(online|offline).json
* 2c10c7d Start with unit tests
* 589b394 Update .gitignore

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+      - name: Remove some files from upload
+        run: |
+           find sphinx_typo3_theme/static -name __init__.py -delete
+           find sphinx_typo3_theme/static -name *_t -delete
+      - name: Upload static as artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: assets-for-cdn
+          path: sphinx_typo3_theme/static
+      - name: TODO Upload to Azure.
```
Comment (benjaminkott) [inline .github/workflows/CI.yml]: we should only deploy on released versions with the full version number.
in previous builds I've also deployed the master branch that was more for internal testing purposes than external usage.

1. it avoids http caching issues
2. we would need to invalidate the caches on the cdn that makes additional work
3. we should not change the appearance of a rendered documentation.
4. this build should run, after we published to pypi
5. the folder that will be uploaded must contain the following structure `theme/sphinx_typo3_theme/<version/branch>` the cdn deployment will simply upload this structure

File: AUTHORS.rst [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
-=====================================
+=======
+Authors
+=======
 
-...
+sphinx_typo3_theme is written and maintained by Martin Bless
+<martin.bless@mbless.de>.
+
+Major developer is Benjamin Kott <benjamin.kott@typo3.org> who created the
```
Comment (benjaminkott) [inline AUTHORS.rst]: please use `benjamin.kott@outlook.com` i do not have a .org mail address.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (marble) [reply .github/workflows/CI.yml]: Ok. Will do. Although I'm regretting that we can't do apply instant changes, for example at patch level, to all manuals.

File: AUTHORS.rst (hunk 2 above)
Comment (marble) [reply AUTHORS.rst]: Thank you for reviewing. I'll change the email.

**Comment by marble** [comment]
@benjaminkott I'm excited to see whether everything works. 'gh-pages' should be updated automatically on push to master. And deploying to PyPi and CDN when we do a release with a valid number. Hopefully! Shall I give it a try an merge?

**Comment by benjaminkott** [comment]
Give it a go! We will fix anything afterwards.

**Pull request merged**